src/bgw/worker.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#ifndef BGW_WORKER_H
+#define BGW_WORKER_H
+
+#include <postgres.h>
+
+#include <postmaster/bgworker.h>
+
+/**
+ * Parameters to background workers.
+ *
+ * Do not add data here that cannot be simply copied to the background worker
+ * using memcpy(3). If it is necessary to add fields that cannot simply be
+ * copied, we need to start using the send and recv functions for the types.
+ */
+typedef struct BgwParams
+{
+	Oid user_oid;
+	int32 job_id; /* Not used for tests */
+	int32 ttl;	  /* Only used for tests */
+
+	/* Reserve some space for additional parameters. */
+	char function[BGW_EXTRALEN - 32];
```

Comment:
A comment on how the value `32` is calculated could be helpful. This would make it easier to adjust the value if the structure needs to be changed in the future.